edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/ServiceProxy.cs

```diff
@@ -12,21 +12,25 @@ namespace Microsoft.Azure.Devices.Edge.Hub.CloudProxy
 
     public class ServiceProxy : IServiceProxy
     {
-        readonly IDeviceScopeApiClient securityScopesApiClient;
+        readonly IDeviceScopeApiClientProvider securityScopesApiClientProvider;
+        readonly IDeviceScopeApiClient defaultSecurityScopesApiClient;
 
-        public ServiceProxy(IDeviceScopeApiClient securityScopesApiClient)
+        public ServiceProxy(IDeviceScopeApiClientProvider securityScopesApiClientProvider)
         {
-            this.securityScopesApiClient = Preconditions.CheckNotNull(securityScopesApiClient, nameof(securityScopesApiClient));
+            this.securityScopesApiClientProvider = Preconditions.CheckNotNull(securityScopesApiClientProvider, nameof(securityScopesApiClientProvider));
+            this.defaultSecurityScopesApiClient = this.securityScopesApiClientProvider.CreateDeviceScopeClient();
         }
 
-        public IServiceIdentitiesIterator GetServiceIdentitiesIterator() => new ServiceIdentitiesIterator(this.securityScopesApiClient);
+        public IServiceIdentitiesIterator GetServiceIdentitiesIterator() => new ServiceIdentitiesIterator(this.defaultSecurityScopesApiClient);
```

Comment:
same, it should only have GetServiceIdentitiesIterator and maybe modify the IDeviceScopeApiClientProvider  to also create iterator. I see the iterator implementation is not public here, but I think it could be made public